juju/juju
# Add machine.WatchApplicationCharmURL to model cache

## Description of change

Add a watcher to model cache to notify when the lxd profiles applied to a machine may need to change.  

## QA steps

1. `export JUJU_DEV_FEATURE_FLAGS=instance-mutater`
2. `bootstrap lxd with --config logging-config='<root>=DEBUG;unit=DEBUG;juju.core.cache=TRACE;juju.worker.instancemuter=TRACE'`
2. `juju debug-log -m controller --replay --include-module juju.worker.instancemutater --include-module juju.core.cache`
3. download the ubuntu charm to a local directory (~/charms)
4. `juju deploy ~/charms/ubunu`
3. the worker should log that a notify was recieved.
3. `juju upgrade-charm ubuntu --path ~/charms/ubuntu`
3. the worker should not log that a notify was recieved.
4. `juju deploy ./testcharms/charm-repo/quantal/lxd-profile`
3. the worker should log that a notify was recieved.
4. `juju upgrade-charm lxd-profile --path ./testcharms/charm-repo/quantal/lxd-profile`
3. the worker should log that a notification was received.
3. `juju add-unit ubuntu --to 1`
4.  the worker should not log that no notification should be received for machine 1.
3. `juju add-unit lxd-profile --to 0`
3. the worker should not log that  notification should be received for machine 0.
